### PR TITLE
Chore(new_chart): Add Chaos Charts for Azure instance terminate experiment

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: udit <udit@chaosnative.com>

#### Issue: 
- https://github.com/litmuschaos/litmus/issues/2418

### Details:

- Add chaos charts for azure instance terminate chaos experiment.
- The Azure instance terminates chaos experiment PowerOff a given instance for a certain chaos duration and start is again after the chaos duration.

### CODE REVIEW

File: charts/azure/azure-instance-terminate/azure-instance-terminate.chartserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - name: Udit Gaurav
+      email: udit@chaosnative.com
+  provider:
+    name: Chaos Native
+  labels:
+    app.kubernetes.io/component: chartserviceversion
+    app.kubernetes.io/version: latest
+  links:
+    - name: Source Code
+      url: https://github.com/litmuschaos/litmus-go/tree/master/experiments/azure/azure-instance-terminate
```
Comment (ispeakc0de) [inline charts/azure/azure-instance-terminate/azure-instance-terminate.chartserviceversion.yaml:35]: https://github.com/litmuschaos/litmus-go/tree/azure/experiments/azure/instance-terminate/experiment

File: charts/azure/azure-instance-terminate/azure-instance-terminate.chartserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    - name: Documentation
+      url: https://docs.litmuschaos.io/docs/azure-instance-terminate/
+    - name: Video
+      url:
+  icon:
+    - url: 
+      mediatype: ""
+  chaosexpcrdlink: https://raw.githubusercontent.com/litmuschaos/chaos-charts/master/charts/azure/azure-instance-terminate/experiment.yaml
```
Comment (ispeakc0de) [inline charts/azure/azure-instance-terminate/azure-instance-terminate.chartserviceversion.yaml:43]: azure-instance-terminate or instance-terminate(experiment-name)?

File: charts/azure/azure-instance-terminate/experiment.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          - "chaosexperiments"
+          - "chaosresults"
+        verbs:
+          - "create"
+          - "list"
+          - "get"
+          - "patch"
+          - "update"
+          - "delete"
+    image: "litmuschaos/go-runner:ci"
```
Comment (ispeakc0de) [inline charts/azure/azure-instance-terminate/experiment.yaml:38]: litmuschaos/go-runner:azure

File: charts/azure/azure-instance-terminate/experiment.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    # provide the target instance name
+    - name: AZURE_INSTANCE_NAME
+      value: ''
+
+    # provide the resource group of the instance
+    - name: RESOURCE_GROUP
+      value: ''
+
+    # Provide the path of aws credentials mounted from secret
```
Comment (ispeakc0de) [inline charts/azure/azure-instance-terminate/experiment.yaml:62]: aws or aks?

File: charts/azure/azure-instance-terminate/rbac.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: azure-instance-terminate-sa
+  labels:
+    name: azure-instance-terminate-sa
+    app.kubernetes.io/part-of: litmus
+rules:
+- apiGroups: ["","litmuschaos.io","batch"]
+  resources: ["pods","jobs","secrets","events","pods/log","pods/exec","chaosengines","chaosexperiments","chaosresults"]
```
Comment (ispeakc0de) [inline charts/azure/azure-instance-terminate/rbac.yaml:20]: Can we split the permissions based on the apiGroups?

File: charts/azure/azure.chartserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  maintainers:
+    - name: ksatchit
+      email: karthik.s@mayadata.io
+  provider:
+    name: Chaos Native  
+  links:
+    - name: Kubernetes Website
+      url: https://kubernetes.io
+    - name: Source Code
+      url: https://github.com/litmuschaos/litmus-go/tree/master/experiments/azure
```
Comment (ispeakc0de) [inline charts/azure/azure.chartserviceversion.yaml:29]: master branch or azure branch?

File: charts/azure/azure.chartserviceversion.yaml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    - name: Kubernetes Slack
+      url: https://slack.kubernetes.io/
+    - name: Documentation
+      url: https://docs.litmuschaos.io/docs/chaoshub/#generic-chaos
```
Comment (ispeakc0de) [inline charts/azure/azure.chartserviceversion.yaml:33]: https://docs.litmuschaos.io/docs/chaoshub/#platform-chaos

File: charts/azure/azure.package.yaml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+packageName: azure 
+experiments:
+  - name: k8-azure-instance-terminate 
+    CSV: k8-azure-instance-terminate.chartserviceversion.yaml
```
Comment (ispeakc0de) [inline charts/azure/azure.package.yaml:4]: do we have k8 prefix?

File: charts/azure/rbac-admin.yaml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: chaos-admin
+  labels:
+    name: chaos-admin
+rules:
+- apiGroups: ["","apps","batch","extensions","litmuschaos.io","openebs.io","storage.k8s.io"]
+  resources: ["chaosengines","chaosexperiments","chaosresults","configmaps","cstorpools","cstorvolumereplicas","events","jobs","persistentvolumeclaims","persistentvolumes","pods","pods/exec","pods/log","secrets","storageclasses","chaosengines","chaosexperiments","chaosresults","configmaps","cstorpools","cstorvolumereplicas","daemonsets","deployments","events","jobs","persistentvolumeclaims","persistentvolumes","pods","pods/eviction","pods/exec","pods/log","replicasets","secrets","services","statefulsets","storageclasses"]
```
Comment (ispeakc0de) [inline charts/azure/rbac-admin.yaml:16]: do we need openebs related permissions? Can we provide minimal permissions?

**Comment by uditgaurav** [comment]
Closing the PR due to outdated forked repo and re-raising it - https://github.com/litmuschaos/chaos-charts/pull/442

**Pull request closed without merging**